modules/core/m_mode.c

```diff
@@ -391,21 +409,14 @@ ms_bmask(struct MsgBuf *msgbuf_p, struct Client *client_p, struct Client *source
 			pbuf += arglen;
 			plen += arglen;
 			modecount++;
+			sprintf(dbuf, "%s ", s);
```

Comment:
You aren't advancing `dbuf` after this, so you're simply overwriting the start of the buffer over and over again.